include/seqan3/io/sequence/sequence_file_format_fasta.hpp

```diff
@@ -170,84 +290,69 @@ class sequence_file_format_fasta
-        { "faa"   },
-        { "frn"   },
-    };
+        if (options.fasta_legacy_id_marker)
+            stream_it = ';';
+        else
+            stream_it = '>';
 
-    //!\copydoc sequence_file_format_concept::read
-    template <typename stream_type,     // constraints checked by file
-              typename options_type,    // given by file
-              typename seq_type,        // other constraints checked inside function
-              typename id_type,
-              typename qual_type,
-              typename seq_qual_type>
-    void read(stream_type        & stream,
-              options_type const & options,
-              seq_type           & sequence,
-              id_type            & id,
-              qual_type          & SEQAN3_DOXYGEN_ONLY(qualities),
-              seq_qual_type      & seq_qual)
-    {
-        static_assert(detail::decays_to_ignore_v<seq_type> || detail::decays_to_ignore_v<seq_qual_type>,
-                      "Either the sequence field, or the seq_qual field need to be set to std::ignore.");
-        static_assert(detail::decays_to_ignore_v<qual_type> || detail::decays_to_ignore_v<seq_qual_type>,
-                      "Either the qualities field, or the seq_qual field need to be set to std::ignore.");
+        if (options.fasta_blank_before_id)
+            stream_it = ' ';
 
-        auto stream_view = ranges::iterator_range{std::istreambuf_iterator<char>{stream},
-                                                  std::istreambuf_iterator<char>{}};
-        // ID
-        read_id(stream_view, options, id);
+        ranges::copy(id, stream_it);
 
-        // Sequence
-        if constexpr (!detail::decays_to_ignore_v<seq_type>) // sequence
-            read_seq(stream_view, options, sequence);
-        else
-            read_seq(stream_view, options, seq_qual);        // seq_qual (possibly std::ignore, too)
+        if (options.add_carriage_return)
+            stream_it = '\r';
+        stream_it = '\n';
+    }
 
-        // make sure "buffer at end" implies "stream at end"
-        if ((std::istreambuf_iterator<char>{stream} == std::istreambuf_iterator<char>{}) &&
-            (!stream.eof()))
+    //!\brief Implementation of writing the sequence.
+    template <typename stream_it_t,
+              typename options_type,
+              typename seq_type>
+    void write_seq(stream_it_t        & stream_it,
+                   options_type const & options,
+                   seq_type          && seq)
+    {
+        if (options.fasta_letters_per_line > 0)
         {
-            stream.get(); // triggers error in stream and sets eof
+            ranges::copy(seq | view::to_char
+                             | ranges::view::chunk(options.fasta_letters_per_line)
+                             | ranges::view::join(options.add_carriage_return
+                                                    ? std::string_view{"\r\n"}
+                                                    : std::string_view{"\n"}),
```

Comment:
question: why not `std::endl`?

Reply:
std::endl is not a string or a range. It is a control data structure. Also it includes `std::flush`. AND we don't want to write whatever the system default is, instead we want to give the user the choice via the options.